Formula/python3.rb

```diff
@@ -310,7 +310,22 @@ def post_install
   end
 
   def xy
-    version.to_s.slice(/(3\.\d)/) || "3.6"
+    # Stable versions
+    ver = version.to_s.slice(/(3\.\d)/)
+
+    if ver.nil?
+        if File.file?("configure.ac")
+            # In Python's source tree
+            ver = File.read("configure.ac").slice(/PYTHON_VERSION, (3\.\d)/, 1)
+        else
+            # Installed packages
+            ver = (prefix/"Frameworks/Python.framework/Versions").children.first.basename.to_s
+        end
+    end
+
+    raise "Cannot determine Python version" unless ver
```

Comment:
Got it. Thanks for all the patience for a Ruby newbie. Refactoring this might take some time :)